badge/src/main/java/net/skyscanner/backpack/badge/BpkBadge.kt

```diff
@@ -0,0 +1,142 @@
+  attrs: AttributeSet?,
+    defStyleAttr: Int) : AppCompatTextView(context, attrs, defStyleAttr) {
+
+    constructor(context: Context) : this(context, null)
+    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, R.style.Bpk_badge)
+
+    init {
+      initialize(context, attrs, defStyleAttr)
+      setup()
+    }
+
+    /**
+     * @property type
+     * Type of badge. Default Type.Success
+     */
+    var type: Type = Type.Success
+      set(value) {
+        field = value
+        setup()
+      }
+    /**
+     * @property message
+     * message on the badge
+     */
+    var message: String? = null
+      set(value) {
+        field = value
+        this.text = message
+      }
+
+    private fun initialize(context: Context, attrs: AttributeSet?, defStyleAttr: Int) {
+
+      val a: TypedArray = context.theme.obtainStyledAttributes(
+        attrs,
+        R.styleable.badge,
+        0, 0)
+
+      if (a.hasValue(R.styleable.badge_type)) {
+        type = Type.fromId(a.getInt(R.styleable.badge_type, 0))
+      }
+      if (a.hasValue(R.styleable.badge_message)) {
+        message = a.getString(R.styleable.badge_message)
+      }
+
+      a.recycle()
+    }
+
+    private fun setup() {
+      if (message != null) {
+        this.text = message
+      }
+
+      if (type == null) {
+        return
+      }
+      //set padding
+      val paddingMd = resources.getDimension(R.dimen.bpkSpacingMd).toInt()
+      val paddingSm = resources.getDimension(R.dimen.bpkSpacingSm).toInt()
+      this.setPadding(paddingMd, paddingSm, paddingMd, paddingSm)
+
+      //set Text color
+      this.setTextColor(context.resources.getColor(type.textColor))
+
+      // Set background color
+      val border = GradientDrawable()
+      border.setColor(context.resources.getColor(type.bgColor))
+
+      //Set border
+      if (type == Type.Outline) {
+        border.setStroke(resources.getDimension(R.dimen.badge_border_size).toInt(), context.resources.getColor(R.color.bpkWhite))
+        //set alpha for border
+        border.setColor(context.resources.getColor(type.bgColor) and 0x32ffffff)
+      }
+
+      //set corner radius
+      val cornerRadius = context.resources.getDimension(R.dimen.bpkBorderRadiusSm)
+
+      val radius = floatArrayOf(cornerRadius, cornerRadius,
+        cornerRadius, cornerRadius,
+        cornerRadius, cornerRadius,
+        cornerRadius, cornerRadius)
+      border.cornerRadii = radius
+      this.background = border
+
+      val params = LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT)
+      this.gravity = Gravity.CENTER
+      this.layoutParams = params
+    }
+
+  enum class Type constructor(internal var id: Int,
+                              internal var type: String,
+                              internal var bgColor: Int,
+                              internal var textColor: Int) {
+    /**
+     * Style for badges with positive messages
```

Comment:
I think for consistency sake we should apply the same rules as the monorepo: comments should be avoided unless strictly necessary, if the code is not clear perhaps can be written in a more explicit way

It's kind of annoying but consistency is key

Reply:
These are javadoc comments. The developers can hover on the function/variable in the IDE to get help on the topic. It's a handy feature and in my opinion we should have it.